search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/LlmInput.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.llm;
+
+/**
+ * Input to LLMs.
+ */
+public interface LlmInput {}
```

Comment:
why interface for all the I/O classes shouldn't we create Abstract class here? Thoughts?

Reply:
Mainly, the reason that I am only exposing interfaces, e.g. `Llm`, to the search pipelines is that I want the flexibility to have different implementations, the OpenSearch Remote Inference remote models being one of them.

As an alternative to the HttpConnector, you could use something like [this](https://github.com/TheoKanning/openai-java) which is a Java client that OpenAI recommends.

As I stated above, I think it's OK to couple this with the Remote Inference framework, but my thinking here is that we get maximal flexibility with a fairly simple design.  I don't think these interfaces overcomplicate things.

Reply:
> I don't think these interfaces overcomplicate things.

Interfaces are good when we know for sure there can be other implementation. Even when there become usecases to do call the LLM without deploying the model, there will still be model entity. We should use that entity here.

Reply:
If you use remote inference endpoints (such as OpenAI, Cohere, SageMaker), you do not need to create model entities in OpenSearch.  Or at least I am arguing that we should not have to force users to create and maintain model entities in OpenSource if all their models are external.  Again, I am fine with having this search pipeline to use a remote model that uses HttpConnector.  I am making a [point](https://github.com/opensearch-project/ml-commons/issues/882#issuecomment-1673814168) that is sort of independent of this PR.

Reply:
@austintlee Where do you want this conversation to happen? I am happy to converse on the issue or here too.

Reply:
I will clean up this PR to only use the remote inference.  We can continue the discussion on whether or not we should force everyone to create models in OpenSearch on #882.